src/core/namespace.c

```diff
@@ -881,12 +883,28 @@ static int mount_procfs(const MountEntry *m, const NamespaceInfo *ns_info) {
         _cleanup_free_ char *opts = NULL;
         const char *entry_path;
         int r, n;
+        struct utsname uts;
+        bool old = false;
 
         assert(m);
         assert(ns_info);
 
-        if (ns_info->protect_proc != PROTECT_PROC_DEFAULT ||
-            ns_info->proc_subset != PROC_SUBSET_ALL) {
+        /* If uname says that the system is older than v5.8, then the textual hidepid= stuff is not
+         * supported by the kernel, and thus the per-instance hidepid= neither, which means we
+         * really don't want to use it, since it would affect our host's /proc * mount. Hence let's
+         * gracefully fallback to a classic, unrestricted version. */
+
+        r = uname(&uts);
+        if (r < 0)
+               return -errno;
```

Comment:
Can uname(2) actually fail for reasons outwith OOM and the like? If so, it would probably be best to just log_debug and skip the strverscmp - failing execution because a spurious log message might appear in dmesg doesn't seem the appropriate reaction

Reply:
we usually assert_se() around it. it can't even really fail due to oom, we already give it all its memory...

it could fail because of seccomp filters i guess, but i don't think we need to cover for that.

Reply:
We're passing the buffer and the only error listed for uname(2) is EFAULT so it seems very unlikely to fail and it if does, there are problem bigger problems about to happen?